### PR TITLE
ACS-7886 Change from Travis to GHA for 7.2.2 service pack

### PR DESCRIPTION
**Pull request by SaraAspery**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  DOCKERHUB_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  GITHUB_ACTIONS_DEPLOY_TIMEOUT: 60
+  MAVEN_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+  MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+  QUAY_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
+  QUAY_USERNAME: ${{ secrets.QUAY_USERNAME }}
+  TAS_ENVIRONMENT: ./tests/environment
+  TAS_SCRIPTS: ../alfresco-community-repo/packaging/tests/scripts
+  # Release version has to start with real version (7.3.0-....) for the docker image to build successfully.
+  RELEASE_VERSION: 7.2.1
```
Comment (pzhyland) [inline .github/workflows/ci.yml:32]: Shouldn't it be 7.2.2?

Comment (pzhyland) [reply .github/workflows/ci.yml:32]: Or even 7.2.2-A1?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            profiles: all-tas-tests
+            compose-file: docker-compose-minimal.yml
+          - testSuite: Integration
+            pom: tests/tas-integration/pom.xml
+            profiles: all-tas-tests
+            compose-file: docker-compose-minimal.yml
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.33.0
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.33.0
```
Comment (pzhyland) [inline .github/workflows/ci.yml:83]: This sets up a Java 17 by default. 7.2.X **need to be compiled with Java 11** Please take a look how it's done here https://github.com/Alfresco/acs-packaging/blob/release/7.2.N/.github/workflows/ci.yml#L18

File: tests/tas-integration/pom.xml [hunk 3]
```diff
@@ -114,8 +114,10 @@
                     <suiteXmlFiles>
                         <suiteXmlFile>${suiteXmlFile}</suiteXmlFile>
                     </suiteXmlFiles>
+                    <!-- Keeping illegal-access=warn for Java 11 compatibility, even though it has no effect on JDK 17 -->
```
Comment (pzhyland) [inline tests/tas-integration/pom.xml:117]: Can it be removed when we switch back to Java11?

File: tests/tas-restapi/pom.xml [hunk 4]
```diff
@@ -84,8 +84,10 @@
                     <suiteXmlFiles>
                         <suiteXmlFile>${suiteXmlFile}</suiteXmlFile>
                     </suiteXmlFiles>
+                    <!-- Keeping illegal-access=warn for Java 11 compatibility, even though it has no effect on JDK 17 -->
```
Comment (pzhyland) [inline tests/tas-restapi/pom.xml:87]: Can it be removed when we switch back to Java11?

**Review by pzhyland — CHANGES_REQUESTED** [review]
Java 11 support is required on the 7.2.N branches

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SaraAspery) [reply .github/workflows/ci.yml:32]: I was planning on changing it to 7.2.2-A1 in the release PR. This PR is just for the conversion to GHA

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+        run: bash ./scripts/ci/cleanup_cache.sh
+
+  community_zip_tests:
+    name: Community Distribution Zip content tests
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[skip tests]')"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.33.0
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.33.0
```
Comment (pzhyland) [inline .github/workflows/ci.yml:108]: Java 11 is needed here.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    if: >
+      !(failure() || cancelled()) &&
+      !contains(github.event.head_commit.message, '[skip docker_latest]') && 
+      (github.ref_name == 'master' || startsWith(github.ref_name, 'release/')) &&
+      github.event_name != 'pull_request'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.33.0
+      - uses: Alfresco/alfresco-build-tools/.github/actions/free-hosted-runner-disk-space@v1.35.2
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.33.0
```
Comment (pzhyland) [inline .github/workflows/ci.yml:135]: Java 11 is needed here.

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+      contains(github.event.head_commit.message, '[release]') &&
+      (github.ref_name == 'master' || startsWith(github.ref_name, 'release/')) &&
+      github.event_name != 'pull_request'
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.33.0
+      - uses: Alfresco/alfresco-build-tools/.github/actions/free-hosted-runner-disk-space@v1.35.2
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.33.0
```
Comment (pzhyland) [inline .github/workflows/ci.yml:162]: Java 11 is needed here.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (SaraAspery) [reply .github/workflows/ci.yml:84]: Done

File: tests/tas-integration/pom.xml (hunk 3 above)
Comment (SaraAspery) [reply tests/tas-integration/pom.xml]: yes

File: tests/tas-restapi/pom.xml (hunk 4 above)
Comment (SaraAspery) [reply tests/tas-restapi/pom.xml]: yes

**Pull request merged**